### PR TITLE
Add NFSN Support

### PR DESCRIPTION
**Pull request by tersers**

### CODE REVIEW

**Comment by CapsuleCD** [comment]
Hi.

I'm an automated pull request bot named [CapsuleCD](https://www.capsulecd.com). I handle testing, versioning and package releases for this project. 
- If you're the owner of this repo, you can click the button below to kick off the CapsuleCD build pipeline and create an automated release.'
- If not, don't worry, someone will be by shortly to check on your pull request. 

[CapsuleCD](https://www.capsulecd.com/project/github/AnalogJ/lexicon/pullrequests/300)

---
If you're interested in learning more about [CapsuleCD](http://www.github.com/AnalogJ/capsulecd), or adding it to your project, you can check it out [here](https://www.capsulecd.com)

File: lexicon/providers/nfsn.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import hashlib
+import json
+import logging
+import random
+import string
+import time
+
+import requests
+
+from .base import Provider as BaseProvider
```
Comment (AnalogJ) [inline lexicon/providers/nfsn.py]:
```suggestion
from lexicon.providers.base import Provider as BaseProvider
```

**Comment by AnalogJ** [comment]
Hey @tersers 
Thanks for the PR! 
Can you also add yourself to the CODEOWNERS file?

Other than that, and a small change to the provider, I think this is good to go.

**Comment by AnalogJ** [comment]
Sorry for the delay getting this PR merged. We're working on updating the configuration system and I want to get that merged before adding new providers.

File: lexicon/providers/nfsn.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+import requests
+
+from lexicon.providers.base import Provider as BaseProvider
+
+from urllib.parse import urlencode
+
+logger = logging.getLogger(__name__)
+
+
```
Comment (adferrand) [inline lexicon/providers/nfsn.py]: Module needs to declare a list named `NAMESERVER_DOMAINS`. It should contain any domain used by the nfsn nameservers.

I suggest the following, based on what the documentation says about the nfsn nameservers:
`NAMESERVER_DOMAINS = ['nearlyfreespeech.net']`

See `lexicon/providers/ovh.py` for implementation example.

**Comment by tersers** [comment]
So does my branch need the additional work? Looks like it was just merged.

**Comment by tersers** [comment]
Awesome, thanks guys!

**Pull request merged**